### ISSUE TITLE
Using java.lang.Object.new.java_class.resource_as_stream(file_path) inside a JAR with JDK 16

### ISSUE THREAD

**Issue by AndyObtiva** [jruby/jruby#6848]
The issue occurs when invoking `java.lang.Object.new.java_class.resource_as_stream(file_path)` from inside a JAR produced by Warbler with JRuby running on JDK 16, but works in JRuby running on JDK 8.

It is tough to provide a reproducible example, so you might have to improvise your own version of it.

**Environment Information**
  
* JDK version (`java --version`): java 16.0.2 2021-07-20 Java(TM) SE Runtime Environment (build 16.0.2+7-67) Java HotSpot(TM) 64-Bit Server VM (build 16.0.2+7-67, mixed mode, sharing)
* JRuby version (`jruby -v`) and command line (flags, JRUBY_OPTS, etc): jruby 9.3.0.0 (2.6.8) 2021-09-17 85c20e780f Java HotSpot(TM) 64-Bit Server VM 16.0.2+7-67 on 16.0.2+7-67 +jit [darwin-x86_64]
* Operating system and platform (e.g. `uname -a`): Darwin Andys-MacBook-Pro.local 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021; root:xnu-7195.81.3~1/RELEASE_X86_64 x86_64

Other relevant info you may wish to add:

Using `java.lang.Object.new.java_class.resource_as_stream(file_path)` from inside a JAR with paths produced by File.expand_path (having uri:classloader:/ prefix)

**Expected Behavior**

When using JRuby with JDK8, this code works when run from inside a JAR file produced by warbler using `java -jar dist/jar_file.jar`:

```ruby
@file_path = File.expand_path('../package/linux/image.png', __dir__) # produces an image path starting with uri:classloader due to being inside a JAR file
if @file_path.start_with?('uri:classloader')
  @jar_file_path = @file_path
  puts '@jar_file_path'
  puts @jar_file_path
  file_path = @jar_file_path.sub(/^uri\:classloader\:/, '').sub('//', '/') # the latter sub is needed for Mac
  puts 'file_path'
  puts file_path
  object = java.lang.Object.new
  file_input_stream = java.lang.Object.new.java_class.resource_as_stream(file_path)
  puts 'file_input_stream'
  puts file_input_stream.inspect
end
```

Output:

```
@jar_file_path
uri:classloader:/snowboard_utah/package/linux/Snowboard Utah.png
file_path
/snowboard_utah/package/linux/Snowboard Utah.png
file_input_stream
#<Java::SunNetWwwProtocolJar::JarURLConnection::JarURLInputStream:0x6f240187>
```

**Actual Behavior**

When using JRuby with JDK 16, I get this output instead (`nil` for `resource_as_stream` call):

Output:

```
@jar_file_path
uri:classloader:/snowboard_utah/package/linux/Snowboard Utah.png
file_path
/snowboard_utah/package/linux/Snowboard Utah.png
file_input_stream
nil
```

**Final Note**

If I am wrong to be using `resource_as_stream` to read a `uri:classloader:/` prefixed path, advice is appreciated.

I admit that my knowledge about JDK 16 is limited compared to JDK 8, so if there is something I need to do due diligence in studying about JDK 16 instead of reporting this issue, then any quick pointers are appreciated.

Thank you.

**Comment by AndyObtiva** [comment]
OK, it seems this is related to the changes that happened in JDK 9 with Modules:
https://stackoverflow.com/questions/47613602/java-9-0-classloadergetresourceasstream-nullpointerexception

I guess I found where I need to do some due diligence. I haven't used JDKs beyond 8 yet, so this is mostly caused by weakness in my knowledge of more modern versions of Java. 

I'll take a look at this tomorrow and see if I could automate a solution for it.

**Comment by AndyObtiva** [comment]
This has been resolved by adding a Java class in a package matching the name of the JRuby project, thus allowing retrieval of any resources from the Warbler-produced JAR file according to uri:classloader paths.

Thanks for continuing to make JRuby available. IT IS TOTALLY AWESOME!!!

On that note, I released [Glimmer DSL for SWT 4.21.0.0](https://github.com/AndyObtiva/glimmer-dsl-swt) with SWT 4.21, the first version of SWT to require a JDK higher than JDK 8 (I recommended JDK 16 in my project since it has the new `jpackage`).

Cheers!

**Comment by AndyObtiva** [comment]
The package in my case is fully mine as it was formed from my project and does not originate in another module, so I believe it is my right to include a generated Java class in it if I want.

Still, I tested the solution you provided, and it worked, so I will utilize instead due to being simpler.

Thank you.
